likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js

```diff
@@ -0,0 +1,234 @@
+        <LikecoinInfoTable
+          likerIdValue={likerIdValue}
+          likerDisplayName={likerDisplayName}
+          likerWalletAddress={likerWalletAddress}
+          likerAvatar={likerAvatar}
+          isLoading={isLoading}
+          handleClickOnChange={handleClickOnChange}
+          handleLikerIdInputChange={handleLikerIdInputChange}
+          isChangingTypingLiker={isChangingTypingLiker}
+          handleDisconnect={handleDisconnect}
+          hasValidLikecoinId={hasValidLikecoinId}
+          editable={!siteLikerIdEnabled}
+          isMainSettingPage={false}
+          showChangeButton={showChangeButton}
+          showDisconnectButton={showDisconnectButton}
+        />
+        <Section title={__('Your Likecoin button', 'likecoin')} />
+        {hasValidLikecoinId && (
+          <LikeButtonPreview
+            userLikerId={likerIdValue}
+            hasValidLikecoinId={hasValidLikecoinId}
+          />
+        )}
+        <SubmitButton />
+      </form>
+    </div>
+  );
+}
+function LikecoinButtonPage() {
+  const {
+    DBSiteLikerId,
+    DBSiteLikerAvatar,
+    DBSiteLikerDisplayName,
+    DBSiteLikerWallet,
+    DBSiteLikerIdEnabled,
+  } = useSelect((select) => select(SITE_LIKER_INFO_STORE_NAME).selectSiteLikerInfo());
+
+  const {
+    DBUserLikerId,
+    DBUserLikerAvatar,
+    DBUserLikerDisplayName,
+    DBUserLikerWallet,
+  } = useSelect((select) => select(USER_LIKER_INFO_STORE_NAME).selectUserLikerInfo());
+
+  const { postUserLikerInfo } = useDispatch(USER_LIKER_INFO_STORE_NAME);
+
+  const [siteLikerIdEnabled] = useState(DBSiteLikerIdEnabled);
+  const [defaultLikerId, setDefaultLikerIdValue] = useState('');
+  const [defaultLikerDisplayName, setDefaultDisplayName] = useState('');
+  const [defaultLikerWalletAddress, setDefaultLikerWalletAddress] = useState('');
+  const [defaultLikerAvatar, setDefaultLikerAvatar] = useState('');
+
+  useEffect(() => {
+    if (siteLikerIdEnabled) {
+      setDefaultLikerIdValue(DBSiteLikerId);
+      setDefaultDisplayName(DBSiteLikerDisplayName);
+      setDefaultLikerWalletAddress(DBSiteLikerWallet);
+      setDefaultLikerAvatar(DBSiteLikerAvatar);
+    } else {
+      setDefaultLikerIdValue(DBUserLikerId);
+      setDefaultDisplayName(DBUserLikerDisplayName);
+      setDefaultLikerWalletAddress(DBUserLikerWallet);
+      setDefaultLikerAvatar(DBUserLikerAvatar);
+    }
+  }, [
+    siteLikerIdEnabled,
+    DBSiteLikerId,
+    DBSiteLikerDisplayName,
+    DBSiteLikerWallet,
+    DBSiteLikerAvatar,
+    DBUserLikerId,
+    DBUserLikerDisplayName,
+    DBUserLikerWallet,
+    DBUserLikerAvatar,
+  ]);
+
+  return (
+    <PureLikecoinButtonPage
+      DBSiteLikerIdEnabled={DBSiteLikerIdEnabled}
+      defaultLikerId={defaultLikerId}
+      defaultLikerDisplayName={defaultLikerDisplayName}
+      defaultLikerWalletAddress={defaultLikerWalletAddress}
+      defaultLikerAvatar={defaultLikerAvatar}
+      postUserLikerInfo={postUserLikerInfo}
+    />
+  );
```

Comment:
No need to use `useState` here as you will not change the state later

```
<PureLikecoinButtonPage
  defaultLikerId={siteLikerIdEnabled ? DBSiteLikerId : DBUserLikerId}
  ...
/>
```

You may also use `useMemo` to cache the value to improve performance 
```
const defaultLikerId = useMemo(
  () => siteLikerIdEnabled ? DBSiteLikerId : DBUserLikerId,
  [
    siteLikerIdEnabled,
    DBSiteLikerId,
    DBUserLikerId,
  ]
)
```

Reply:
Cool that it can be simplified like this. Changed below.